syntax/abstract.mjs

```diff
@@ -1,36 +1,52 @@
 import * as FTL from "./ast.mjs";
+import {always, never} from "../lib/combinators.mjs";
```

Comment:
This lays a foundation for adding more semantics checks to `abstract.mjs`. By making `into` return a new `Parser` instance itself, we can chain parsers (a.k.a. _flatmap_, `then` or `and_then`) and reject parse results during the AST construction.

Reply:
Would these combinators read better as `accept` and `reject`?

Reply:
Those don't read significantly better to me. I wouldn't want two names for the same thing, though, and in grammar.mjs, the current terminology seems to be more fitting.